### DISCUSSION TITLE
How do I add the listed abbrev as completion items?

### DISCUSSION THREAD

**Discussion by pimvh** [olets/zsh-abbr#92] [General]
Hi all,

First of all, thanks for providing this tool open source! 

This might be slightly out-of-scope for this repo, or an unsupported use-case, if so, let me know.

I was wondering whether anyone is using this is combination with the built-in completion of zsh. 'normal' aliases (I think natively?) already show up and I would like to have zsh-abbr 'aliases' also show up. 

I have searched the documentation but have not been able to find an example documenting this.

I am looking for something like the following:

Currently completion of arguments (as well as the loved `zsh-autosuggestions` in light-gray): 
<img width="854" alt="Screenshot 2023-05-13 at 15 59 39" src="https://github.com/olets/zsh-abbr/assets/17687712/c0478964-491f-4ec1-853a-7e237e7edff8">

Desired behaviour:
```
(prompt) ➜ ls -
 -- option --
-1      -- single column output
.
...
<< other arguments >
-- zsh-abbr --
ls     -- long-command-i-created-an-alias-ls-for
```

I hope my example has been clear, if not, please let me know.

NB: I know using the `ls` command as an `zsh-abbr` is probably a dumb idea, it's just for illustration.

**Comment by olets** [comment]
The alias behavior is new to me. Can you give an example?

**Reply by pimvh** [reply]
Sure!|

I am using zsh 5.9, and have the following completion function available: 

```
➜ cat _aliases
#compdef unalias

local expl sel args opts

zparseopts -E -D s:=sel

[[ -z $sel ]] && sel=rgs

opts=( "$@" )

args=()
[[ $sel = *r* ]] && args=( $args 'aliases:regular alias:compadd -k aliases' )
[[ $sel = *g* ]] && args=( $args 'global-aliases:global alias:compadd -k galiases' )
[[ $sel = *s* ]] && args=( $args 'suffix-aliases:suffix alias:compadd -k saliases' )
[[ $sel = *R* ]] && args=( $args 'disabled-aliases:disabled regular alias:compadd -k dis_aliases' )
[[ $sel = *G* ]] && args=( $args 'disabled-global-aliases:disabled global alias:compadd -k dis_galiases' )
[[ $sel = *S* ]] && args=( $args 'disabled-suffix-aliases:disabled suffix alias:compadd -k dis_saliases' )

_alternative -O opts $args
```

I am not sure on the details on how zsh makes this completion available to me, but it eventually ends up in my shell.

If I set my .zshrc to the following (probably also including some statement which load the completion definition):

`.zshrc`
```
alias vim=nvim
```

Then, when I type the following in my prompt, and list native completions:
```
-> vim
```

I get the following result: 

```
➜ vim
 -- external command --
vim       vimdiff   vimtutor
 -- alias --
vim
```
I thus know that it is aliases to something else.

In the same way, it might be useful to list `zsh-abbr` aliases in the native completion syntax, by providing a completion definition. I do not know how to achieve this, bringing me to my original question.

**Reply by olets** [reply]
That's neat, and new to me. If you find a solution please share it!
